### PR TITLE
Style Variations: Filter out the colors and fonts

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/issues/97134

## Proposed Changes

* Don't show palette variations and typography variations on the Theme page to align with the Core. In addition, 
  * The slug between style variations and color variations might be the same, and it made the selection weird
  * The preview of the font variations look too similar

| Before | After | Core |
| - | - | - |
| image | image | image |

## Why are these changes being made?

* https://github.com/Automattic/wp-calypso/issues/97134

## Testing Instructions

* Go to /theme/twentytwentyfive
* Ensure there are only 8 style variations

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/global-styles/src/gutenberg-bridge/index.tsx [hunk 1]
```diff
@@ -64,10 +65,67 @@ const useSafeGlobalStylesOutput = () => {
...
+	);
+
+	return areGlobalStyleConfigsEqual( variationWithProperties, variation );
+};
+
+const isColorVariation = ( variation: GlobalStylesObject ) =>
+	isVariationWithProperties( variation, [ 'color' ] );
+
+const isFontVariation = ( variation: GlobalStylesObject ) =>
+	isVariationWithProperties( variation, [ 'typography' ] );
```
Comment (arthur791004) [inline packages/global-styles/src/gutenberg-bridge/index.tsx:121]: Align with [TypographyVariations](https://github.com/WordPress/gutenberg/blob/4ad26e1e90735b80fa390f9fe6dd1832e50b876b/packages/edit-site/src/components/global-styles/variations/variations-typography.js#L18) in the Gutenberg.

File: packages/global-styles/src/gutenberg-bridge/index.tsx [hunk 2]
```diff
@@ -64,10 +65,67 @@ const useSafeGlobalStylesOutput = () => {
...
+	const variationWithProperties = filterObjectByProperties(
+		window.structuredClone( variation ),
+		properties
+	);
+
+	return areGlobalStyleConfigsEqual( variationWithProperties, variation );
+};
+
+const isColorVariation = ( variation: GlobalStylesObject ) =>
+	isVariationWithProperties( variation, [ 'color' ] );
```
Comment (arthur791004) [inline packages/global-styles/src/gutenberg-bridge/index.tsx:118]: Align with [ColorVariations](https://github.com/WordPress/gutenberg/blob/4ad26e1e90735b80fa390f9fe6dd1832e50b876b/packages/edit-site/src/components/global-styles/variations/variations-color.js#L18) in the Gutenberg.

**Review by fushar — COMMENTED** [review]
The theme card still shows 20+ variations, should we also fix that 🤔 

image

**Review by fushar — APPROVED** [review]
Tested TT5 and other themes.

I think my comment above is not a blocker.

**Comment by arthur791004** [comment]
> The theme card still shows 20+ variations, should we also fix that 🤔

Looks like we have to remove badges there. See https://github.com/Automattic/wp-calypso/issues/93931

**Pull request merged**